### DISCUSSION TITLE
How to pass on --data-binary  to jason api query

### DISCUSSION THREAD

**Discussion by harryamin** [grafana/grafana-infinity-datasource#306] [Specific APIs]
HI Ram,
This is my curl and its working.

curl https://api.newrelic.com/graphql \
  -H 'Content-Type: application/json' \
  -H 'API-Key: xxxxxxxxxxxxxxxxxxxxxx' \
  --data-binary '{"query":"{\n  actor {\n    account(id: 1234567) {\n      nrql(query: \"SELECT apdex(apm.service.apdex.user) AS EndUserHealthScore FROM Metric facet appName\") {\n        nrql\n        rawResponse\n      }\n    }\n  }\n}\n", "variables":""}'

how do I pass on --data-binary? any help or suggestion is highly  appreciated.

**Comment by harryamin** [comment]
After looking  other example of placing the query in Method and Body as POST, I tried this as well ...it did not work(I have Api-Key and Content-Type define in Headers.
{
  actor {
    account(id: 1234567) {
      nrql(query: "SELECT apdex(apm.service.apdex.user) AS endUserHealthScore FROM Metric facet appName") {
        nrql
        rawResponse
      }
    }
  }
}

**Reply by yesoreyeram** [reply]
What you needed is graphql here
